server/src/main/java/org/opensearch/search/query/QuerySearchResult.java

```diff
@@ -364,6 +367,11 @@ public void readFromWithId(ShardSearchContextId id, StreamInput in) throws IOExc
         nodeQueueSize = in.readInt();
         setShardSearchRequest(in.readOptionalWriteable(ShardSearchRequest::new));
         setRescoreDocIds(new RescoreDocIds(in));
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            tookTimeNanos = new SetOnce<>(in.readOptionalLong());
+        } else {
+            tookTimeNanos = new SetOnce<>();
+        }
```

Comment:
```suggestion
        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
            tookTimeNanos.set(in.readOptionalLong());
        }
```

Reply:
That is a very good point. I definitely should have thought of that... the wrapper concept came in much later as we were worried about reading the whole QSR into memory on each eviction, and by then I didn't consider going back and changing this. In that case, we can probably close this PR and just stick that computation into the upcoming policies PR.